Kafka/src/main/java/io/deephaven/kafka/ingest/KafkaIngester.java

```diff
@@ -161,19 +152,22 @@ public String toString() {
     /**
      * Creates a Kafka ingester for all partitions of a given topic.
      *
-     * @param log a log for output
-     * @param props the properties used to create the {@link KafkaConsumer}
-     * @param topic the topic to replicate
-     * @param partitionToConsumer a function implementing a mapping from partition to its consumer of records.
-     * @param partitionToInitialSeekOffset a function implementing a mapping from partition to its intial seek offset,
+     * @param log A log for output
+     * @param props The properties used to create the {@link KafkaConsumer}
+     * @param topic The topic to replicate
+     * @param partitionToStreamConsumer A function implementing a mapping from partition to its consumer of records. The
```

Comment:
You are saying the implementations should not allocate resources, but I do not think that is actually true of what the implementations do.

Reply:
I mean they definitely do stuff by creating the adapter and inserting stuff into a table.   The question about whether the comment or code is "right" is somewhat philosophical.

                final Pair<StreamToTableAdapter, ConsumerRecordToStreamPublisherAdapter> partitionAdapterPair =
                        adapterFactory.get();
                final Table partitionTable = tableConversion.apply(partitionAdapterPair.getFirst().table());
                streamTableMap.enqueueUpdate(() -> Assert.eqNull(streamTableMap.put(partition, partitionTable),
                        "streamTableMap.put(partition, partitionTable)"));
                return new SimpleKafkaStreamConsumer(partitionAdapterPair.getSecond(), partitionAdapterPair.getFirst());

Reply:
Well, the comment has an "if appropriate" hedge. In this case, it's my intent to create the adapters and do the table map puts. Do you think I should re-word it somehow to make it clearer that its programmer's-choice? Existing comment pasted here for reference.
```
     * @param partitionToStreamConsumer A function implementing a mapping from partition to its consumer of records. The
     *        function will be invoked once per partition at construction; implementations should internally defer
     *        resource allocation until first call to {@link KafkaStreamConsumer#accept(Object)} or
     *        {@link KafkaStreamConsumer#acceptFailure(Exception)} if appropriate.
```